### PR TITLE
TAG-14 Search all href of profiles

### PR DESCRIPTION
**Pull request by pawelsitarz98**

### CODE REVIEW

File: searchscript/search_names.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+parent = document;
+function GetPersonID(xpath, parent)
+{ 
+    let results = [];
+    let query = document.evaluate(xpath,parent,null,XPathResult.ORDERED_NODE_SNAPSHOT_TYPE,null);
+    for (let i = 0, length = query.snapshotLength; i < length; ++i) {
+        results.push(query.snapshotItem(i));
+    }
+    return results;
+
```
Comment (orzeech) [inline searchscript/search_names.js:11]: remove this empty line

File: searchscript/search_names.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+xpath = "//a[@class='app-aware-link feed-shared-actor__container-link relative display-flex flex-grow-1']/@href";
+parent = document;
+function GetPersonID(xpath, parent)
```
Comment (orzeech) [inline searchscript/search_names.js:3]: this does not return ID but href

File: searchscript/search_names.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+xpath = "//a[@class='app-aware-link feed-shared-actor__container-link relative display-flex flex-grow-1']/@href";
```
Comment (orzeech) [inline searchscript/search_names.js:1]: this xpath is too concrete - make it more idiotproof

**Review by orzeech — CHANGES_REQUESTED** [review]

**Comment by orzeech** [comment]
Make this work with our extension, after clicking on "Search" or something -> print all hrefs to console

File: searchscript/search_names.js (hunk 2 above)
Comment (orzeech) [reply searchscript/search_names.js:3]: in js function names start with small letter

**Review by orzeech — CHANGES_REQUESTED** [review]

File: searchscript/search_names.js (hunk 3 above)
Comment (pawelsitarz98) [reply searchscript/search_names.js:1]: Fixed, now contain three situations and more flexible implementation

File: searchscript/searchNames.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+document.addEventListener('DOMContentLoaded', function() {
+  var checkPageButton = document.getElementById('searchNames');
+  checkPageButton.addEventListener('click', function() {  
+      function searchNames() {
+          console.log('List of people:');       
+          xpath = "//a[contains(@class,'actor') or contains(@class,'entity-type-card') or contains(@class,'ember-view msg-conversation')]/@href"; //version for homepage
```
Comment (pawelsitarz98) [inline searchscript/searchNames.js:6]: //Version for homepage, mynetwork and messaging

File: extension/popup.html [hunk 5]
```diff
@@ -30,11 +31,22 @@
    .button2:hover {  
       background-color: #008CBA;  
       color: white;  
-      }  
+      }
+   .button3{
+      background-color: white;  
+      color: black;  
+      border: 2px solid #c22b05;
+   }
```
Comment (orzeech) [inline extension/popup.html:39]: before merging use consistent formatting here

**Review by orzeech — APPROVED** [review]

**Pull request merged**